### DISCUSSION TITLE
building an web application that take screenshot out of the browser

### DISCUSSION THREAD

**Discussion by HarshadHindlekar** [community/community#42182] [Repositories]
### Select Topic Area

Product Feedback

### Body

hey is that possible that an web application have functionality to take screenshot of pc window whether user is not on that browser but that application is running on that browser 
Ex.    user is login to an application which has functionality to take screenshot in a perticular time interval  whether user open youtube, vs code or an application on their machine/pc. 

is their possible then plz let me know .

thank you.

**Comment by getlost01** [comment]
It is possible for a web application to take a screenshot of a user's desktop or a specific window while the user is using other applications, but it would require the user to grant the web application permission to do so. This can be done using browser APIs such as the `getUserMedia()` API in combination with the MediaStream API, which allows the web application to access the user's camera and microphone.

To take a screenshot of the desktop or a specific window, the web application would need to capture the user's screen or window using the `getUserMedia()` API and then display the captured image on the web page. This would require the user to grant the web application permission to access their screen or window.

It is also worth noting that taking a screenshot of the user's desktop or a specific window without the user's knowledge or consent may be considered a security or privacy risk, and it is generally not recommended to do so. It is important for web developers to be transparent about their use of such features and to obtain the user's explicit consent before accessing sensitive information.

**Reply by HarshadHindlekar** [reply]
yeah i  understand developers policy but my prj build for an organization they have that requirement .
can you tell me how to use getUserMedia() to take screenshot
because i use this to capture webCam images but i dont know how to use it to take screenshot.

thank you

**Reply by airtower-luna** [reply]
> It is also worth noting that taking a screenshot of the user's desktop or a specific window without the user's knowledge or consent may be considered a security or privacy risk, and it is generally not recommended to do so.

That's a nice way to say "highly illegal in many jurisdictions". :wink:
